### PR TITLE
feat(CX-1713) Add search by artwork title

### PR DESCRIPTION
**Pull request by Gon4ar5**
Resolves [CX-1713](https://artsyproduct.atlassian.net/browse/CX-1713)

Expected Outcome
Convection to automatically update sales results from Ohm, which would give all teams better and faster data as well

First, I added a field for setting source_artwork_id directly from convection (this field is used only when creating a submission via volt, at the moment there are only 186 such submissions).
After I added an consignment endpoint, which sends a gravity request about a closed sale and updates the prices.

If you know someone else who can review🤔, please add to reviewers😅 |

### CODE REVIEW

File: app/controllers/api/consignments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Api
+  class ConsignmentsController < RestController
+    before_action :require_authentication
+    before_action :set_submission, only: %i[update]
+
+    def update
+      consignment = @submission.consigned_partner_submission
```
Comment (Gon4ar5) [inline app/controllers/api/consignments_controller.rb:9]: sending submission_id in url, looks bad

File: app/controllers/api/consignments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      sale_artworks = Gravity.client.sales(id: id).sale_artworks
+
+      sale_artworks.map do |sale_artwork|
+        artwork_id = sale_artwork.artwork.id
+        price = sale_artwork.highest_bid.try(:amount_cents)
+        [artwork_id: artwork_id, price: price]
+      end
+    end
+
+    def consignment_params
```
[the remark this answers is not in the document]
Comment (Gon4ar5) [reply app/controllers/api/consignments_controller.rb:36]: relic of the past

File: app/controllers/api/consignments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Api
+  class ConsignmentsController < RestController
+    before_action :require_authentication
+
+    def update
+      # id in request == sale_id
+      artworks_price = fetch_artworks_price(params[:id])
```
[the remark this answers is not in the document]
Comment (Gon4ar5) [reply app/controllers/api/consignments_controller.rb:9]: sounds good

[the remark this answers is not in the document]
Comment (Gon4ar5) [reply app/controllers/api/consignments_controller.rb:9]: but what artworks should fetch_artworks return? if fetch_artworks returns sale_artworks, then I don't think that the method makes sense at all. If it returns artworks, then we still need to call sale_artwork in order to find out the price

[the remark this answers is not in the document]
Comment (olerichter00) [reply app/controllers/api/consignments_controller.rb:9]: Yeah, you're right, `fetch_sale_artworks` makes a lot of sense to me.

[the remark this answers is not in the document]
Comment (Gon4ar5) [reply app/controllers/api/consignments_controller.rb:9]: There was a little misunderstanding😄. 
if I call the method fetch_sale_artworks, then I will expect it to return sale_artworks
it happens here `Gravity.client.sales(id: id).sale_artworks`
so the method is no longer needed

**Review by bhoggard — APPROVED** [review]

File: app/controllers/api/consignments_controller.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (olerichter00) [reply app/controllers/api/consignments_controller.rb:9]: Would `fetch_sale_artworks_with_price` make sense? We can also call it `fetch_artworks_price`, it just confused me and I had to look up what this method actually returns.

**Review by olerichter00 — APPROVED** [review]
Great work - feel free to merge 🙌

**Pull request merged**